app/src/main/java/ch/epfl/qedit/view/home/HomeQuizListFragment.java

```diff
@@ -21,39 +27,88 @@
 public class HomeQuizListFragment extends Fragment {
     public static final String QUIZID = "ch.epfl.qedit.view.QUIZID";
 
-    private ListView listView;
+    private CustomAdapter customAdapter;
+    private User user;
+
+    private HomePopUp homePopUp;
 
     @Override
     public View onCreateView(
             LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         // Inflate the layout for this fragment
         final View view = inflater.inflate(R.layout.fragment_home_quiz_list, container, false);
-        listView = view.findViewById(R.id.home_quiz_list);
+        RecyclerView recyclerView = view.findViewById(R.id.home_quiz_list);
 
         // Get user from the bundle created by the parent activity
         final User user = (User) Objects.requireNonNull(getArguments()).getSerializable("user");
+        this.user = user;
 
-        ArrayList<Map.Entry<String, String>> entries =
-                new ArrayList<>(user.getQuizzes().entrySet());
-
-        final CustomAdapter adapter = new CustomAdapter(requireActivity(), entries);
+        this.customAdapter = new CustomAdapter(requireActivity());
+        recyclerView.setAdapter(customAdapter);
+        recyclerView.setLayoutManager(new GridLayoutManager(getActivity(), 1));
 
-        listView.setAdapter(adapter);
-        listView.setOnItemClickListener(
-                new AdapterView.OnItemClickListener() {
-                    @Override
-                    public void onItemClick(
-                            AdapterView<?> parent, View view, int position, long id) {
-                        Map.Entry<String, String> item =
-                                (Map.Entry<String, String>) adapter.getItem(position);
+        recyclerView.addItemDecoration(
+                new DividerItemDecoration(getContext(), DividerItemDecoration.VERTICAL));
 
-                        startQuizActivity(item.getKey());
-                    }
-                });
+        homePopUp = new HomePopUp(getContext(), this.user, customAdapter);
+        // Have to set it to true to show the menu, if the user is an editor
+        if (user.getRole() == User.Role.Editor | user.getRole() == User.Role.Administrator) {
```

Comment:
why do we need a bit-wise or? I mean, it works for boolean but it's kind of weird right?

Reply:
I don't see what you mean

Reply:
I think it might be a typo. You used `|` instead of `||`.

Reply:
same thing